RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalRecHitProducerPortable.cc

```diff
@@ -0,0 +1,179 @@
+        recHitsTokenEB_{produces(ps.getParameter<std::string>("recHitsLabelEB"))},
+        recHitConditionsToken_{esConsumes()},
+        recHitParametersToken_{esConsumes()} {
+    if (!isPhase2_) {
+      recHitsTokenEE_ = produces(ps.getParameter<std::string>("recHitsLabelEE"));
+    }
+    configParameters_.killDeadChannels = ps.getParameter<bool>("killDeadChannels");
+    configParameters_.EBLaserMIN = ps.getParameter<double>("EBLaserMIN");
+    configParameters_.EELaserMIN = isPhase2_ ? 0. : ps.getParameter<double>("EELaserMIN");
+    configParameters_.EBLaserMAX = ps.getParameter<double>("EBLaserMAX");
+    configParameters_.EELaserMAX = isPhase2_ ? 0. : ps.getParameter<double>("EELaserMAX");
+
+    // do not propagate channels with these flags on
+    uint32_t flagmask = 0;
+    flagmask |= 0x1 << EcalRecHit::kNeighboursRecovered;
+    flagmask |= 0x1 << EcalRecHit::kTowerRecovered;
+    flagmask |= 0x1 << EcalRecHit::kDead;
+    flagmask |= 0x1 << EcalRecHit::kKilled;
+    flagmask |= 0x1 << EcalRecHit::kTPSaturated;
+    flagmask |= 0x1 << EcalRecHit::kL1SpikeFlag;
+    configParameters_.flagmask = flagmask;
+
+    // for recovery and killing
+    configParameters_.recoverEBIsolatedChannels = ps.getParameter<bool>("recoverEBIsolatedChannels");
+    configParameters_.recoverEEIsolatedChannels =
+        isPhase2_ ? false : ps.getParameter<bool>("recoverEEIsolatedChannels");
+    configParameters_.recoverEBVFE = ps.getParameter<bool>("recoverEBVFE");
+    configParameters_.recoverEEVFE = isPhase2_ ? false : ps.getParameter<bool>("recoverEEVFE");
+    configParameters_.recoverEBFE = ps.getParameter<bool>("recoverEBFE");
+    configParameters_.recoverEEFE = isPhase2_ ? false : ps.getParameter<bool>("recoverEEFE");
+  }
+
+  void EcalRecHitProducerPortable::produce(device::Event& event, device::EventSetup const& setup) {
+    auto& queue = event.queue();
+
+    // get device collections from event
+    auto const* uncalibRecHitsDevEB = &event.get(uncalibRecHitsTokenEB_);
+    auto const* uncalibRecHitsDevEE = isPhase2_ ? nullptr : &event.get(uncalibRecHitsTokenEE_);
+
+    // get the size of the input collections from the metadata
+    auto const uncalibRecHitsSizeEB = uncalibRecHitsDevEB->const_view().metadata().size();
+    auto const uncalibRecHitsSizeEE = isPhase2_ ? 0 : uncalibRecHitsDevEE->const_view().metadata().size();
+
+    // output device collections with the same size than the input collections
+    auto recHitsDevEB = std::make_unique<OutputProduct>(uncalibRecHitsSizeEB, queue);
+    auto recHitsDevEE =
+        isPhase2_ ? std::unique_ptr<OutputProduct>() : std::make_unique<OutputProduct>(uncalibRecHitsSizeEE, queue);
+    // reset the size scalar of the SoA
+    // memset takes an alpaka view that is created from the scalar in a view to the portable device collection
+    auto recHitSizeViewEB =
+        cms::alpakatools::make_device_view<uint32_t>(alpaka::getDev(queue), recHitsDevEB->view().size());
```

Comment:
it should also be possible to write
```suggestion
        cms::alpakatools::make_device_view<uint32_t>(queue, recHitsDevEB->view().size());
```

Reply:
This does not compile. `error: no matching function for call to 'make_device_view<uint32_t>(alpaka::QueueGenericThreadsBlocking<alpaka::DevCpu>&, unsigned int&)'`

Reply:
Mhm, could you try with a newer release ?
It should work with CMSSW_14_2_0_pre2 and later.